cadc-datalink-server/src/main/java/org/opencadc/datalink/DataLink.java

```diff
@@ -0,0 +1,177 @@
+*
+*  This file is part of the             Ce fichier fait partie du projet
+*  OpenCADC project.                    OpenCADC.
+*
+*  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+*  you can redistribute it and/or       vous pouvez le redistribuer ou le
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*  $Revision: 5 $
+*
+************************************************************************
+ */
+
+package org.opencadc.datalink;
+
+import ca.nrc.cadc.dali.tables.votable.VOTableField;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * A single result output by the data link service. Exactly one of accessURL, serviceDef,
+ * and errorMessage must be set prior to output.
+ *
+ * @author pdowler
+ */
+public class DataLink {
+
+    /**
+     * Terms from the http://www.ivoa.net/rdf/datalink/core vocabulary plus
+     * CAOM extensions.
+     */
+    public enum Term { // TODO: re-use the VocabularyTerm code once extracted from caom2
+        THIS("#this"),
+        AUXILIARY("#auxiliary"),
+        PREVIEW("#preview"),
+        PROGENITOR("#progenitor"),
+        DERIVATION("#derivation"),
+        PROC("#proc"),
+        CUTOUT("#cutout"),
+        THUMBNAIL("http://www.openadc.org/caom2#thumbnail"), // TODO: move these out
+        PKG("http://www.openadc.org/caom2#pkg");             // TODO: move these out
```

Comment:
There is a typo in the hostname.
`openadc` should be `opencadc`.

Reply:
fixed